File: cakk-domain/src/main/java/com/cakk/domain/repository/jpa/BusinessInformationJpaRepository.java
package com.cakk.domain.repository.jpa;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.cakk.domain.entity.user.BusinessInformation;

public interface BusinessInformationJpaRepository extends JpaRepository<BusinessInformation, Long> {

	@Query("select bi from BusinessInformation as bi join fetch bi.cakeShop "
		+ "where bi.cakeShop.id =:cakeShopId")
	Optional<BusinessInformation> findBusinessInformationWithCakeShop(@Param("cakeShopId") Long cakeShopId);

	Optional<BusinessInformation> findBusinessInformationByCakeShopId(Long cakeShopId);
}

File: cakk-infrastructure/persistence-mysql/src/main/java/com/cakk/infrastructure/persistence/repository/jpa/BusinessInformationJpaRepository.kt
package com.cakk.infrastructure.persistence.repository.jpa

import com.cakk.common.enums.VerificationStatus
import com.cakk.infrastructure.persistence.entity.user.BusinessInformationEntity
import com.cakk.infrastructure.persistence.entity.user.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface BusinessInformationJpaRepository : JpaRepository<BusinessInformationEntity, Long> {

	@Query(
		"select bi from BusinessInformationEntity as bi join fetch bi.cakeShop "
            + "where bi.cakeShop.id =:cakeShopId")
    fun findBusinessInformationWithCakeShop(@Param("cakeShopId") cakeShopId: Long): BusinessInformationEntity?

	fun findAllByUser(userEntity: UserEntity): List<BusinessInformationEntity>

	fun existsBusinessInformationByUserAndCakeShop_Id(owner: UserEntity, cakeShopId: Long): Boolean

    @Query("select bi from BusinessInformationEntity as bi join fetch bi.cakeShop where bi.user =:user")
    fun findAllWithCakeShopByUser(user: UserEntity): List<BusinessInformationEntity>

    @Query(
		"select bi from BusinessInformationEntity as bi join fetch bi.user join fetch bi.cakeShop"
            + " where bi.verificationStatus =:verificationStatus")
    fun findAllCakeShopBusinessOwnerCandidates(verificationStatus: VerificationStatus): List<BusinessInformationEntity>

    @Query(
		"select bi from BusinessInformationEntity as bi join fetch bi.user join fetch bi.cakeShop"
            + " where bi.user.id =:userId")
    fun findBusinessInformationByUserId(userId: Long): BusinessInformationEntity?
}

